domain/application/service/application.go

```diff
@@ -775,6 +781,40 @@ func (s *Service) SetUnitPassword(ctx context.Context, unitName coreunit.Name, p
 	})
 }
 
+// GetUnitWorkloadStatus returns the workload status of the specified unit, returning an
+// error satisfying [applicationerrors.UnitNotFound] if the unit doesn't exist.
+func (s *Service) GetUnitWorkloadStatus(ctx context.Context, unitName coreunit.Name) (*corestatus.StatusInfo, error) {
+	if err := unitName.Validate(); err != nil {
+		return nil, errors.Trace(err)
+	}
+	unitUUID, err := s.st.GetUnitUUIDByName(ctx, unitName)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	agentStatus, err := s.st.GetUnitWorkloadStatus(ctx, unitUUID)
```

Comment:
Can't `GetUnitWorkloadStatus()` take a `coreunit.Name` as param instead of a UUID? It's just a different `WHERE` clause and the method is only used here.

Reply:
See below